agent/api/task/service_connect_test.go

```diff
@@ -0,0 +1,261 @@
+	bridgeMode                      = "bridge"
+	testAwsvpcPortOverride          = "8080"
+	testAwsvpcPortDefault           = "8080"
+	testBridgePortOverride          = "8080"
+	testBridgePortDefault           = "15000"
+	testBridgeHostPort              = "8080"
+	testServiceConnectContainerName = "ecs-service-connect"
+	testStatsConfigPath             = "testPath"
+	testHostName                    = "testHostName"
+	testAddress                     = "testAddress"
+	testOutboundListenerName        = "testOutboundListener"
+	testInboundListenerName         = "testInboundListener"
+	testIPV4Cidr                    = "testIPV4Cidr"
+	testIPV6Cidr                    = "testIPV6Cidr"
+)
+
+func getTestACSTaskAttachment(networkMode string, override bool) *ecsacs.Task {
+	ingressConfig := ""
+	switch networkMode {
+	case awsvpcMode:
+		if override {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s`, testAwsvpcPortOverride)
+		} else {
+			ingressConfig = fmt.Sprintf(`\"interceptPort\":%s,\"listenerName\":\"%s\"`, testAwsvpcPortDefault, testInboundListenerName)
+		}
+	case bridgeMode:
+		if override {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s,\"hostPort\":%s`, testBridgePortOverride, testBridgeHostPort)
+		} else {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s`, testBridgePortDefault)
+		}
+	}
+
+	testAWSVPCServiceConnectConfig := fmt.Sprintf(`"{\"statsConfig\":{\"path\":\"%s\"},\"egressConfig\":{\"listenerName\":\"%s\",\"vip\":{\"ipv4Cidr\":\"%s\",\"ipv6Cidr\":\"%s\"}},\"dnsConfig\":[{\"hostname\":\"%s\",\"address\":\"%s\"}],\"ingressConfig\":[{%s}]}"`, testStatsConfigPath, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testHostName, testAddress, ingressConfig)
+
+	return &ecsacs.Task{
+		Attachments: []*ecsacs.Attachment{
+			{
+				AttachmentArn: strptr("attachmentArn"),
+				AttachmentProperties: []*ecsacs.AttachmentProperty{
+					{
+						Name:  strptr(serviceConnectAttProSCContainerName),
+						Value: strptr(testServiceConnectContainerName),
+					},
+					{
+						Name:  strptr(serviceConnectAttProSCConfigName),
+						Value: strptr(testAWSVPCServiceConnectConfig),
+					},
+				},
+				AttachmentType: strptr(serviceConnectAttType),
+			},
+		},
+	}
+}
+
+func getTestServiceConnectConfig(scHostName, scAddress, scOutListenerName, scIPV4Cidr, scIPV6Cidr, scPath string, scIngressConfig []IngressConfigEntry) *ServiceConnectConfig {
+	return &ServiceConnectConfig{
+		ContainerName: testServiceConnectContainerName,
+		StatsConfig: StatsConfig{
+			Path: scPath,
+		},
+		IngressConfig: scIngressConfig,
+		EgressConfig: EgressConfig{
+			ListenerName: scOutListenerName,
+			VIP: VIP{
+				IPV4CIDR: scIPV4Cidr,
+				IPV6CIDR: scIPV6Cidr,
+			},
+		},
+		DNSConfig: []DNSConfigEntry{
+			{
+				HostName: scHostName,
+				Address:  scAddress,
+			},
+		},
+	}
+}
+
+func TestExtractSCConfigFromACSAttachment(t *testing.T) {
+	tt := []struct {
+		testName      string
+		ingressConfig []IngressConfigEntry
```

Comment:
nit: can we rename this to `ExpectedIngressConfig`

Reply:
ingressConfig is renamed to expectedIngressConfig in commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054. Thanks!